app/services/service_listeners/attachment_updater.rb

```diff
@@ -1,6 +1,13 @@
 module ServiceListeners
   class AttachmentUpdater
+    extend LockedDocumentConcern
+
     def self.call(attachable: nil, attachment_data: nil)
+      if attachable.is_a?(Edition)
+        document = attachable.document
+        check_if_locked_document_with_document_id(document.id)
```

Comment:
Since we're checking the item is an edition it might be slightly simpler to follow if we pass the edition into check.